dirty_cat/test/test_super_vectorizer.py

```diff
@@ -18,90 +19,171 @@ def check_same_transformers(expected_transformers: dict, actual_transformers: li
     vectorizer_base = SuperVectorizer(
-        cardinality_threshold=3,
+        cardinality_threshold=4,
         # we must have n_samples = 5 >= n_components
         high_card_str_transformer=GapEncoder(n_components=2),
         high_card_cat_transformer=GapEncoder(n_components=2),
         numerical_transformer=StandardScaler(),
     )
     # Warning: order-dependant
-    expected_transformers_df = {
-        'numeric': ['int', 'float'],
-        'low_card_str': ['str1'],
-        'high_card_str': ['str2'],
-        'low_card_cat': ['cat1'],
-        'high_card_cat': ['cat2'],
-    }
     vectorizer_base.fit_transform(X)
     check_same_transformers(expected_transformers_df, vectorizer_base.transformers)
 
     # Test with higher cardinality threshold and no numeric transformer
     vectorizer_default = SuperVectorizer()  # Using default values
-    expected_transformers_2 = {
-        'low_card_str': ['str1', 'str2'],
-        'low_card_cat': ['cat1', 'cat2'],
-    }
     vectorizer_default.fit_transform(X)
     check_same_transformers(expected_transformers_2, vectorizer_default.transformers)
 
     # Test with a numpy array
     arr = X.to_numpy()
     # Instead of the columns names, we'll have the column indices.
-    expected_transformers_np = {
-        'numeric': [0, 1],
-        'low_card_str': [2, 4],
-        'high_card_str': [3, 5],
-    }
     vectorizer_base.fit_transform(arr)
-    check_same_transformers(expected_transformers_np, vectorizer_base.transformers)
+    check_same_transformers(expected_transformers_np_no_cast, vectorizer_base.transformers)
 
     # Test with pandas series
-    expected_transformers_series = {
-        'low_card_cat': ['cat1'],
-    }
     vectorizer_base.fit_transform(X['cat1'])
     check_same_transformers(expected_transformers_series, vectorizer_base.transformers)
 
     # Test casting values
     vectorizer_cast = SuperVectorizer(
-        cardinality_threshold=3,
+        cardinality_threshold=4,
         # we must have n_samples = 5 >= n_components
         high_card_str_transformer=GapEncoder(n_components=2),
         high_card_cat_transformer=GapEncoder(n_components=2),
         numerical_transformer=StandardScaler(),
     )
     X_str = X.astype('object')
-    expected_transformers_plain = {
-        'high_card_str': ['str2', 'cat2'],
-        'low_card_str': ['str1', 'cat1'],
-        'numeric': ['int', 'float']
-    }
     # With pandas
     vectorizer_cast.fit_transform(X_str)
     check_same_transformers(expected_transformers_plain, vectorizer_cast.transformers)
     # With numpy
     vectorizer_cast.fit_transform(X_str.to_numpy())
-    check_same_transformers(expected_transformers_np, vectorizer_cast.transformers)
+    check_same_transformers(expected_transformers_np_cast, vectorizer_cast.transformers)
 
 
+def test_with_clean_data():
+    X = _get_clean_dataframe()
+    # Define the transformers we'll use throughout the test.
```

Comment:
Tell us a bit more in plain English in the comments how you built these list, so that someone coming here in a few years understands the logic of the test :)